src/services/item/plugins/app/appData/service.ts

```diff
@@ -56,12 +69,15 @@ export class AppDataService {
     body: Partial<InputAppData>,
   ) {
     const { appDataRepository, memberRepository, itemRepository } = repositories;
+
     // check member exists
     if (!actorId) {
       throw new MemberCannotWriteItem();
     }
     const member = await memberRepository.get(actorId);
 
+    await this.hooks.runPreHooks('post', member, repositories, { appData: body, itemId });
```

Comment:
Should this be moved after the checks (e.g line 103 instead)?

Reply:
I agree.